sys/auto_init/saul/auto_init_lis2dh12.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2018 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ *
+ */
+
+/**
+ * @ingroup     auto_init_saul
+ * @{
+ *
+ * @file
+ * @brief       Auto initialization of LIS2DH12 accelerometers
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#ifdef MODULE_LIS2DH12
+
+#include "log.h"
+#include "saul_reg.h"
+#include "lis2dh12.h"
+#include "lis2dh12_params.h"
+
+/**
+ * @brief   Define the number of configured sensors
+ */
+#define LIS2DH12_NUM    (sizeof(lis2dh12_params) / sizeof(lis2dh12_params[0]))
+
+/**
+ * @brief   Allocate memory for the device descriptors
+ */
+static lis2dh12_t lis2dh12_devs[LIS2DH12_NUM];
+
+/**
+ * @brief   Memory for the SAUL registry entries
+ */
+static saul_reg_t saul_entries[LIS2DH12_NUM];
+
+void auto_init_lis2dh12(void)
+{
+    for (unsigned int i = 0; i < LIS2DH12_NUM; i++) {
```

Comment:
Maybe add an assert between LIS2DH12_NUM and a LIS2DH12_INFO_NUM, see #7937 for examples.